stdlib/universe/filter.go

```diff
@@ -140,174 +139,50 @@ func (s *FilterProcedureSpec) PlanDetails() string {
 	return "<non-Expression>"
 }
 
-type filterTransformation struct {
-	execute.ExecutionNode
-	d               *execute.PassthroughDataset
-	ctx             context.Context
-	fn              *execute.RowPredicateFn
-	keepEmptyTables bool
-	alloc           *memory.Allocator
-}
-
 func NewFilterTransformation(ctx context.Context, spec *FilterProcedureSpec, id execute.DatasetID, alloc *memory.Allocator) (execute.Transformation, execute.Dataset, error) {
 	fn := execute.NewRowPredicateFn(spec.Fn.Fn, compiler.ToScope(spec.Fn.Scope))
 	t := &filterTransformation{
-		d:               execute.NewPassthroughDataset(id),
-		fn:              fn,
 		ctx:             ctx,
+		fn:              fn,
 		keepEmptyTables: spec.KeepEmptyTables,
-		alloc:           alloc,
-	}
-
-	if feature.NarrowTransformationFilter().Enabled(ctx) {
-		t := &filterTransformationAdapter{
-			filterTransformation: t,
-		}
-		return execute.NewNarrowTransformation(id, t, alloc)
 	}
-	return t, t.d, nil
+	return execute.NewNarrowTransformation(id, t, alloc)
 }
 
-func (t *filterTransformation) RetractTable(id execute.DatasetID, key flux.GroupKey) error {
-	return t.d.RetractTable(key)
+type filterTransformation struct {
+	ctx             context.Context
+	fn              *execute.RowPredicateFn
+	keepEmptyTables bool
 }
 
-func (t *filterTransformation) Process(id execute.DatasetID, tbl flux.Table) error {
+func (t *filterTransformation) Process(chunk table.Chunk, d *execute.TransportDataset, mem arrowmem.Allocator) error {
 	// Prepare the function for the column types.
-	cols := tbl.Cols()
+	cols := chunk.Cols()
 	fn, err := t.fn.Prepare(cols)
 	if err != nil {
 		// TODO(nathanielc): Should we not fail the query for failed compilation?
 		return err
 	}
 
-	// Retrieve the inferred input type for the function.
-	// If all of the inferred inputs are part of the group
-	// key, we can evaluate a record with only the group key.
-	if t.canFilterByKey(fn, tbl) {
-		return t.filterByKey(tbl)
-	}
-
 	// Prefill the columns that can be inferred from the group key.
 	// Retrieve the input type from the function and record the indices
 	// that need to be obtained from the columns.
 	record := values.NewObject(fn.InputType())
-	indices := make([]int, 0, len(tbl.Cols())-len(tbl.Key().Cols()))
-	for j, c := range tbl.Cols() {
-		if idx := execute.ColIdx(c.Label, tbl.Key().Cols()); idx >= 0 {
-			record.Set(c.Label, tbl.Key().Value(idx))
+	indices := make([]int, 0, len(chunk.Cols())-len(chunk.Key().Cols()))
```

Comment:
It might be difficult. With table chunks, there is a chance it doesn't output the schema in the exact same way. The storage and retrieval also has a cost. I would like to explore making storage and retrieval easier though. If storage and retrieval of schemas are consistently fast and don't take too much memory, we could massage the schema in more places and do more optimizations like this.

I think we could reuse the slice though. That might be helpful.